lib/src/main/java/graphql/nadel/validation/NadelSchemaValidationError.kt

```diff
@@ -430,21 +431,32 @@ sealed interface NadelSchemaValidationError {
         val sourceFieldName: String,
         val sourceFieldTypeName: String,
         val predicateTypeName: String,
-        val overallField: GraphQLFieldDefinition
+        val overallField: GraphQLFieldDefinition,
     ) : NadelSchemaValidationError {
-        override val message = "When condition source field \"${sourceFieldName}\" of type \"${sourceFieldTypeName}\" " +
-            "does not match the predicate type ${predicateTypeName} in the when condition"
+        override val message =
+            "When condition source field \"${sourceFieldName}\" of type \"${sourceFieldTypeName}\" " +
+                "does not match the predicate type ${predicateTypeName} in the when condition"
         override val subject = overallField
     }
 
     data class WhenConditionPredicateRequiresStringSourceField(
         val sourceFieldName: String,
         val sourceFieldTypeName: String,
         val predicateType: String,
-        val overallField: GraphQLFieldDefinition
+        val overallField: GraphQLFieldDefinition,
+    ) : NadelSchemaValidationError {
+        override val message =
+            "When condition source field \"${sourceFieldName}\" of type \"${sourceFieldTypeName}\" " +
+                "needs to be of type String or ID in order to use the ${predicateType} predicate."
+        override val subject = overallField
+    }
+
+    data class SomeHydrationsHaveMissingConditions(
+        val parentType: NadelServiceSchemaElement,
+        val overallField: GraphQLFieldDefinition,
     ) : NadelSchemaValidationError {
-        override val message = "When condition source field \"${sourceFieldName}\" of type \"${sourceFieldTypeName}\" " +
-            "needs to be of type String or ID in order to use the ${predicateType} predicate."
+        val of = makeFieldCoordinates(parentType.overall.name, overallField.name)
```

Comment:
This should not be a property. Refer to all the other message builders, they use `run` to limit the scope of variables used in making the message.